src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs

```diff
@@ -101,7 +105,13 @@ public async Task SynchronizeWithServer()
 
                 var taintIssueVizs = taintVulnerabilities.Select(converter.Convert).ToArray();
                 taintStore.Set(taintIssueVizs);
-                UpdateTaintIssuesUIContext(taintIssueVizs.Any());
+
+                var hasTaintIssues = taintIssueVizs.Length > 0;
+                UpdateTaintIssuesUIContext(hasTaintIssues);
+                if (hasTaintIssues)
+                {
+                    toolWindowService.EnsureToolWindowExists(TaintToolWindow.ToolWindowId);
```

Comment:
Delay creating the tool window as long as possible. It will only be created in situations where we actually want to update the caption i.e. there are taint issues.

Reply:
Why do we need to do this if we're setting the UI context?
It should flip the `ProvideToolWindowVisibility` switch and show the toolwindow - isn't it the same as calling `EnsureToolWindowExists`?

Reply:
No. VS will draw the tool window tab without actually creating the window content if it can e.g. if the tab isn't top-most.
In that case, the tab will appear with whatever caption it had last time VS shut down, and it won't change dynamically.

I'll add a comment.